servlet/src/main/java/io/undertow/servlet/api/WebXMLParser.java

```diff
@@ -0,0 +1,629 @@
+package io.undertow.servlet.api;
+
+import io.undertow.jsp.JspFileHandler;
+import io.undertow.jsp.JspServletBuilder;
+import io.undertow.servlet.core.ManagedServlet;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.EventListener;
+import java.util.HashMap;
+import java.util.LinkedList;
+import javax.servlet.DispatcherType;
+import javax.servlet.Filter;
+import javax.servlet.MultipartConfigElement;
+import javax.servlet.Servlet;
+import javax.servlet.descriptor.JspConfigDescriptor;
+import javax.servlet.descriptor.JspPropertyGroupDescriptor;
+import javax.servlet.descriptor.TaglibDescriptor;
+import javax.servlet.http.HttpServlet;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.apache.jasper.runtime.InstanceManagerFactory;
+import org.apache.jasper.servlet.JspServlet;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class WebXMLParser {
+    protected interface WebXMLParserFactory {
+        void parse(DeploymentInfo deploymentInfo, Element element) throws ClassNotFoundException, WebXMLException;
```

Comment:
If anything this should use STAX parser not DOM, as using DOM slows down boot for at least 10x times.